src/sortStudents.ts

```diff
@@ -1,16 +1,62 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  let aAverage;
+  let bAverage;
+
+  const getAverage = (student: Student): number => student.grades
+    .reduce((acc, grade) => acc + grade) / student.grades.length;
+
+  return [...students].sort((firstEl, secondEl) => {
+    let a = firstEl;
+    let b = secondEl;
+
+    if (order === 'desc') {
+      a = secondEl;
+      b = firstEl;
+    }
+
+    switch (sortBy) {
+      case SortType.Name:
+        return a.name.localeCompare(b.name);
+
+      case SortType.Surname:
+        return a.surname.localeCompare(b.surname);
```

Comment:
Can you convert elements on SortType to lowercase and use here sortBy?

```suggestion
      case SortType.Name:
      case SortType.Surname:
        return a[sortBy].localeCompare(b[sortBy]);
```